### ISSUE TITLE
Calendar selection widget

### ISSUE THREAD

**Issue by Tishka17** [Tishka17/aiogram_dialog#10]
* [x] Single date
* [ ] Date range
* [x] Minimal/Maximal date limitation
* [x] Multiselect - see [mega example](https://github.com/Tishka17/aiogram_dialog/blob/develop/example/mega/bot_dialogs/calendar.py#L58)

**Comment by Tishka17** [comment]
old version https://paste.ubuntu.com/p/zHwXGdgyqG/

**Comment by Tishka17** [comment]
https://paste.ubuntu.com/p/pW3ms49Nbr/

**Comment by MyZik** [comment]
Idea 1: translatable labels for a Calendar widget. I think it can be implemented with a [gettext](https://docs.python.org/3/library/gettext.html) package.

The structure would then look like this:
```
aiogram_dialog/
docs/
example/
translations/
├─ de/
│  ├─ calendar.po
├─ en/
│  ├─ calendar.po
├─ fr/
│  ├─ calendar.po
├─ ru/
│  ├─ calendar.po
```

And in translations/en/calendar.po (for example) we have strings:
```
msgid "Mon"
msgstr ""

msgid "Tue"
msgstr ""

msgid "Wed"
msgstr ""

msgid "Thu"
msgstr ""

msgid "Fri"
msgstr ""

msgid "Sat"
msgstr ""

msgid "Sun"
msgstr ""
```

Translation to ru locale:
```
msgid "Пн"
msgstr ""

msgid "Вт"
msgstr ""

msgid "Ср"
msgstr ""

msgid "Чт"
msgstr ""

msgid "Пт"
msgstr ""

msgid "Сб"
msgstr ""

msgid "Вс"
msgstr ""
```

Idea 2: it would be nice, if we can start from a specific date.
```python
class Calendar(Keyboard, ABC):
    def __init__(self,
                 id: str,
                 start_date: Union[datetime] = None
                 on_click: Union[OnDateSelected, WidgetEventProcessor, None] = None,
                 when: Union[str, Callable] = None,
                 ):
        super().__init__(id, when)
        self.on_click = ensure_event_processor(on_click)
        
        ...
        
window_reservation_at = Window(
    Format(_("Choose the reservation date")),
    Calendar(id='reservation_at',  start_date: datetime.now(), on_click=reservation_at_handler),
    Row(cancel_button),
    state=UserBotTableReservationSG.reservation_at,
)
```

**Comment by Tishka17** [comment]
1. Translations should done using `babel` package or calendar python module with proper locale set. You do not need to create your own translations. 
2. Not sure wht is the meaning of `start_date`. Is it default date, minimal date or what?

**Comment by MyZik** [comment]
> Not sure wht is the meaning of start_date. Is it default date, minimal date or what?

Yes. If a minimal date `20.09.2021`, we can only select the date, that is greater than `20.09.2021` :)

**Comment by vlkorsakov** [comment]
The current view of the Calendar widget is:
![изображение](https://user-images.githubusercontent.com/85243265/226210947-05bbfe8a-6d01-4d5f-b4df-9b7187c266fc.png) 

I suggest splitting the calendar into several widgets:
![изображение](https://user-images.githubusercontent.com/85243265/226211550-23b9499b-375d-4814-9cde-792319e4ad9d.png)

1st lvl:
- `ZoomOut` - Raises the zoom level by one from the current level, starting at level 1 and up to level 3.
- `Week` - Displays the days of the week and allows them to be positioned either above, below, or not displayed at all.
- `Month` or `Calendar` - Displays the days of the month, and on click, returns a `datetime.date` object in the `on_click` func.
- `PrevMonth` and `NextMonth` - No comments.

2nd lvl:
- `Months` - Allows the user to remove any months.

3rd lvl:
- `Years` - Can be customized to allow the user to specify range, such as passing `range(2000, 2024)`, or a selector that will get the `range` from the getter.

While this will kill backward compatibility, but we can it since we are still in beta.

**Comment by Tishka17** [comment]
Implemented highly customzable widget in #265

Though selection is not yet supported.

**Comment by Tishka17** [comment]
Added an example for multiselect - https://github.com/Tishka17/aiogram_dialog/blob/develop/example/mega/bot_dialogs/calendar.py#L58

Date range selection looks very complicated from UX side, so probably won't be implemented.
